lib/oli_web/components/delivery/actions/actions.ex

```diff
@@ -194,6 +213,110 @@ defmodule OliWeb.Components.Delivery.Actions do
     {:noreply, assign(socket, user_role_id: previous_role_id)}
   end
 
+  # Transfer data to another section
+
+  def handle_event("transfer_data_modal", _params, socket) do
+    options_for_select = Sections.get_sections_to_transfer_data(socket.assigns.section)
+
+    modal_assigns = %{
+      title: "Transfer data",
+      id: "transfer_data_modal",
+      ok:
+        JS.push("ok_transfer",
+          target: socket.assigns.myself
+        ),
+      cancel:
+        JS.push("cancel_transfer_modal",
+          target: socket.assigns.myself
+        ),
+      options_for_select: options_for_select
+    }
+
+    modal = fn assigns ->
+      ~F"""
+        <SelectSection.render {...@modal_assigns}/>
+      """
+    end
+
+    send(self(), {:show_modal, modal, modal_assigns})
+
+    {:noreply, assign(socket, modal_assigns: nil)}
+  end
+
+  def handle_event("ok_transfer", %{"section_id" => section_id_to_transfer}, socket) do
+    section_id_to_transfer = String.to_integer(section_id_to_transfer)
+    section_to_transfer = Sections.get_section!(section_id_to_transfer)
+
+    %{
+      section: section,
+      user: user,
+      enrollment: enrollment
+    } = socket.assigns
+
+    result =
+      Repo.transaction(fn ->
+        with {:ok, %Enrollment{} = new_enrollment} <-
+               Sections.enroll(user.id, section_id_to_transfer, [
+                 ContextRoles.get_role(:context_learner)
+               ]),
+
+             ## Update payments for current enrollment with new enrollment id
+             {changes_count, nil} when is_integer(changes_count) <-
+               Paywall.update_payments_for_enrollment(
+                 enrollment,
+                 new_enrollment.id,
+                 section_id_to_transfer
+               ),
+             ## Delete current enrollment
+             {:ok, _enrollment} <- Sections.delete_enrollment(enrollment),
+
+             ## Delete attempts and resource accesses for section to transfer
+             {:ok, _} <- Core.delete_attempts_and_resource_accesses(section_to_transfer, user.id),
+             ## Delete snapshots for section to transfer
+             {changes_count, nil} when is_integer(changes_count) <-
+               Snapshots.delete_snapshots_by_section(section_id_to_transfer),
```

Comment:
This will have dire, unintended side effects.  `Snapshots.delete_snapshots_by_section` as currently written will delete **ALL** snapshot records for the section when the intent is to delete only the snapshot records for that user in that section. `Snapshots.update_snapshots_by_section` has the same problem.